platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -139,19 +172,50 @@ If invalid content is breaking the production build, and therefore blocking the
+- **VA.gov users**: Maintain existing time to see updated content and risk of seeing invalid content
+- **Anybody trying to ship code for the VA.gov front end**: Reduce failed deploys which should allow them to ship faster
 
 ## Specifics
 
 ### Detailed Design
 
-Using the current content build script as a reference, we will write another build script that only includes the steps necessary for the content validation. We will reuse the existing broken link check from the current content build. And using the existing accessibility tests as a reference, we will configure the accessibility tests to not stop on failure so we can identify all the accessibility errors. The script will generate a report of all the broken links and all the accessibility errors. Those reports will be sent to the #cms-team channel, just like the current broken link notifications today.
+#### New build script to supplement existing build script
+
+Before axe and broken link checks can be run, the content needs to be built. A new build script will be created to supplement the existing build script. The new build script will be derived from the existing script, and modified to only include the steps necessary for the content validation. Since the existing build script uses Metalsmith, the new build script will also use Metalsmith.
+
+#### Reuse existing `check-broken-link` plugin in new build script
+
+We will reuse the existing `check-broken-link` Metalsmith plugin in the new build script.
+
+#### Update accessibility tests config
+
+The [current accessibility tests](https://github.com/department-of-veterans-affairs/vets-website/blob/f93e0c11d9e037bd2460ede5e2fb4b67fbaaf118/Jenkinsfile#L82-L84) use Nightwatch. The default value for [Nightwatch's `end_session_on_fail` test session setting](https://nightwatchjs.org/gettingstarted/configuration/#test-session-settings) is true. Since we want a report of all the failures, we would need to update the `end_session_on_fail` setting in our [Nightwatch config](https://github.com/department-of-veterans-affairs/content-build/blob/master/config/nightwatch.docker-compose.js) from `true` to `false`.
+
+#### Improved invalid content reporting
+
+Currently, accessibility errors need to be manually communicated from the Frontend Tools team to the CMS team. And broken links need to be extracted from the build log. We would improve that reporting by generating a file using Winston, and send that file to the #cms-team channel using existing Slack integrations.
+
+#### New scheduled job in Jenkins
 
-Lastly, we will write a scheduled Jenkins workflow to run the content validation script every workday at midnight.
+Lastly, we will write a scheduled Jenkins workflow to run the content validation script every workday at midnight Eastern. That way, the CMS team should have an accurate report of invalid content when they start their workday.
```

Comment:
how long do we anticipate this job taking? could we run it more frequently? (hourly? every 10 minutes?)

Reply:
It looks like the existing `e2e` and `accessibility` steps, which run in parallel, take about 16 minutes to run. There's no reason I can currently think of why we couldn't run the job more frequently. I think hourly makes sense. Do you have any concerns about noise from an increased frequency?

Reply:
The tradeoff seems to be "increased noise" vs "increased time to catch a Veteran-facing issue", right?

If that's the case, I'm inclined to lean more toward "increased noise". We can always come up with creative ways to deal with noise (i.e. some extra logic to avoid alerting multiple times about the same issue), not as much for user-visible accessibility issues on the site waiting to be caught by validation.